store/store.go

```diff
@@ -765,3 +765,120 @@ func (s *SnapUbuntuStoreRepository) SuggestedCurrency() string {
 	}
 	return s.suggestedCurrency
 }
+
+// Optionally specify parameters to the store purchase.
+// Currency: ISO 4217 code as string
+// BackendID: The payment backend to use, e.g. "credit_card", "paypal".
+// MethodID: The unique identifier for the payment method, retrieved from the pay server.
+type BuyOptions struct {
+	Currency  string
+	BackendID string
+	MethodID  int64
+}
+
+// When the purchase state is "InProgress" (i.e. requires user interaction to complete)
+// this struct contains the information required to complete the purchase.
```

Comment:
```
// BuyRedirect holds information required to complete the purchase when state is InProgress,
// case in which it requires user interaction to complete.
```

That said, how about renaming it to `BuyResult` and adding a `State` field on it?  This way we can keep all the data contained, and also handle further cases that may need more data without additional result arguments.